src/fastfield/reader.rs

```diff
@@ -75,29 +104,35 @@ pub enum DynamicFastFieldReader<Item: FastValue> {
 
 impl<Item: FastValue> DynamicFastFieldReader<Item> {
     /// Returns correct the reader wrapped in the `DynamicFastFieldReader` enum for the data.
-    pub fn open(file: FileSlice) -> crate::Result<DynamicFastFieldReader<Item>> {
-        let mut bytes = file.read_bytes()?;
-        let id = bytes.read_u8();
-
+    pub fn open_from_id(
+        bytes: OwnedBytes,
+        id: u8,
+        gcd: u64,
```

Comment:
hmmm so you choose to make gcd somewhat special... Could it have been just another Codec? (well wrapping another codec)

Reply:
Yes, currently it's its own layer, e.g.

Create: i64 -> convert to u64 -> find/apply gcd -> detect/serialize codec
Read: read u64 from codec -> apply gcd -> convert to i64

When we move it into the codec, every codec needs to be gcd aware.

On the read side, the codecs are already wrapped in `FastFieldReaderCodecWrapper`. We could have a separate wrapper like `FastFieldReaderCodecGCDWrapper`, but that would be a performance optimization.

Reply:
No I was suggesting moving it as one codec.
```
- codec1
- codec2
- gcd<C: Codec>
```

Reply:
on the read path that is.